stan/math/rev/mat/fun/cholesky_decompose.hpp

```diff
@@ -321,19 +322,22 @@ class cholesky_opencl : public vari {
    *
    */
   virtual void chain() {
-    auto L_adj_cpu = Eigen::MatrixXd::Zero(M_, M_).eval();
-    auto L_cpu = Eigen::MatrixXd::Zero(M_, M_).eval();
+    int packed_size = M_ * (M_ + 1) / 2;
+    std::vector<double> L_adj_cpu(packed_size);
+    std::vector<double> L_val_cpu(packed_size);
```

Comment:
Looking [here](https://stackoverflow.com/questions/8928547/vector-initialization-or-reserve) not a big difference, but may be better to declare these with `std::vector<double> L_val_cpu;` then use `L_val_cpu.reserve(packed_size);` to avoid some construction costs of the underlying type.

Reply:
That makes total sense. Fixed.

Reply:
I think you can only use the reserve pattern with subsequent push_backs, see http://www.gotw.ca/gotw/074.htm

Reply:
specifically this part: 
> In this case, we used v.reserve(2) and therefore know that v.capacity() >= 2, and that's fine as far as it goes -- but we never actually added any elements to v, so v is still empty! At this point v just happens to have room for two or more elements. We can only safely use operator[]() (or at()) to modify elements that are actually present, which means that they count towards size(). At first you might wonder why operator[]() couldn't just be smart enough to add the element if it's not already there, but if operator[]() were allowed to work this way, you could create a vector with "holes" in it!

Reply:
Thanks Sean. I will switch to using push_back then rather than reverting back to avoid the construction costs.